library.js

```diff
@@ -289,7 +289,11 @@ plugin.addMiddleware = function(req, res, next) {
 	function handleGuest (req, res, next) {
 		if (plugin.settings.guestRedirect && !req.originalUrl.startsWith(nconf.get('relative_path') + '/login?local=1')) {
 			// If a guest redirect is specified, follow it
-			res.redirect(plugin.settings.guestRedirect.replace('%1', encodeURIComponent(nconf.get('url') + req.originalUrl)));
+			var redirectUrl = req.session.returnTo ? req.session.returnTo : req.originalUrl;
```

Comment:
@Buyachaka should not have a `req.session && req.session.returnTo ? ....`  here? in case req.session is null.